pkg/lightning/backend/backend.go

```diff
@@ -197,6 +198,14 @@ type AbstractBackend interface {
 
 	// LocalWriter obtains a thread-local EngineWriter for writing rows into the given engine.
 	LocalWriter(ctx context.Context, cfg *LocalWriterConfig, engineUUID uuid.UUID) (EngineWriter, error)
+
+	// CollectLocalDuplicateRows collect duplicate keys from local db. We will store the duplicate keys which
+	//  are produced by `duplicateIter`.
+	CollectLocalDuplicateRows(ctx context.Context, tbl table.Table, sqlMode mysql.SQLMode) error
```

Comment:
what is a `duplicateIter` in the context of this backend-agnostic interface

Reply:
Because this interface only use for `local` backend. I just means the duplicate key collected in local. Do you think I shall remove it?